source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1437,21 +1437,12 @@ SwiftLanguageRuntime::MetadataPromise::FulfillTypePromise(Status *error) {
       m_remote_ast->getTypeForRemoteTypeMetadata(
           swift::remote::RemoteAddress(m_metadata_location));
 
-  if (result) {
-    m_compiler_type = CompilerType(m_swift_ast, result.getValue().getPointer());
-    if (log)
-      log->Printf("[MetadataPromise] result is type %s",
-                  m_compiler_type->GetTypeName().AsCString());
-    return m_compiler_type.getValue();
-  } else {
-    const auto &failure = result.getFailure();
-    if (error)
-      error->SetErrorStringWithFormat("error in resolving type: %s",
-                                      failure.render().c_str());
-    if (log)
-      log->Printf("[MetadataPromise] failure: %s", failure.render().c_str());
-    return (m_compiler_type = CompilerType()).getValue();
-  }
+  assert(result && "RemoteAST answer is invalid!");
```

Comment:
Do you want user stack traces when this fails in release mode? If so, we should use lldbassert() here.

Otherwise this lgtm, as I don't see much benefit from logging the failure cause, and the code removal is good.

Reply:
So this is definitely not a fatal error, and is one that we need to handle so an assert is not appropriate.  For instance if swift ever were to lazily initialize some generic variable, it's quite likely that when we try to print it it's type is currently unknown.  At that point returning an empty CompilerType is entirely appropriate.

Reply:
Presently it appears we don't have a test for this case, and the point of this patch is to make it easier to find these cases. Once we do, we can return an invalid CompilerType here with more confidence, or perhaps change the API to return an Optional<CompilerType> to eliminate issues which arise when using a type without checking IsValid().